### DISCUSSION TITLE
Disable Update checker for FOSS flavour

### DISCUSSION THREAD

**Discussion by shuvashish76** [d4rken-org/sdmaid-se#753] [General]
As per F-Droid criteria, requires `Tracking` [Anti-Feature](https://f-droid.org/en/docs/Anti-Features/#Tracking) flag if
> Checking for updates without your knowledge or permission.
---
AFAIK Izzy repo also follows F-Droid standard Anti-Feature flags hence applicable to there as well

I'm not a regular "sdmaid-se" user, just installed slightly older version it does show me new "Update available" notification. Connects to `api.github.com` and `raw.githubusercontent.com`.

Suggestion:
i. Disable Internet permission completely: Is there any specific feature/functionality (other than update checker) that requires internet permission? If no then I'd suggest to remove it since checking_update/downloader is the job of App Stores not the app itself. But that's just **my opinion** .
ii. Opt-in Checker: If you do want to have this feature, can you make it opt-in i.e. Disable by default?

There are few more of your apps on Izzy repo, could you disable update checker, _if enabled by default_? Note that F-Droid use AFs for transparency & flags used from the users' point of view.
Thanks

**Comment by d4rken** [comment]
Yes the FOSS build does check some data that is stored on GitHub.

It checks against releases in the repo for updates, and there is a MOTD system that checks for a potential message to be displayed to the user.

It currently can't be disabled, but that's planned. I plan to add a toggle to the settings and on the privacy page during onboarding.

The default will be "enabled" though, as users will forget to enable it and then contact me with issues from outdated versions. There is a non-neglible amount of users that just get the releases straight from GitHub, without an AppStore.

TL;DR
Making this optional: Yes.
Making this opt-in: No.

**Reply by d4rken** [reply]
An issue to track progress of adding the toggle: #755



Side note: The MOTD check can already be disabled, but not from the privacy page during onboarding, I should change that :thinking: :+1: . #754

**Comment by IzzySoft** [comment]
> As per F-Droid criteria

… an auto-updater that's enabled by default is not allowed (as it bypasses F-Droid's security checks), see inclusion criteria. It is tolerated when it's strictly opt-in and makes the implications clear (especially that those updates do not come from F-Droid).

Just for clarification – I see you're already working on it, thanks!

**Comment by d4rken** [comment]
Should be resolved with the next update, see #755
